organise changelog and bump version to 3.0.0dev (#3137)
nf-core/tools: CHANGELOG.md

# nf-core/tools: Changelog

## v3.0.0dev

**Highlights**

> - Pipeline commands are renamed from nf-core <comand> to nf-core pipelines <command>

Reviewer: ```suggestion
- Pipeline commands are renamed from nf-core <comand> to nf-core pipelines <command>, so the follow the same command structure like modules and subworkflow commands
```

> - More customisation for pipeline templates

Reviewer: ```suggestion
- More customisation for pipeline templates, e.g. you can create a new pipeline without any traces of FastQC in it.
```

> - A Text User Interface app when running nf-core pipelines create

Reviewer: ```suggestion
- A Terminal User Interface app when running `nf-core pipelines create` to help us guide you through the process better (no worries, you can still use the cli if you give all values as parameters)
```

> - The pipeline template will come with the nf-schema plugin

Reviewer: ```suggestion
- We replaced nf-validation with nf-schema in the pipeline template
```

> - New command nf-core pipelines rocrate to create a Research Object (RO) crate for a pipeline

Reviewer: ```suggestion
- New command `nf-core pipelines ro-crate` to create a [Research Object (RO) crate](https://www.researchobject.org/ro-crate/) for a pipeline
```

- `nf-core licences` command is deprecated.

> - Pipeline linting will run with the used version of the template

Reviewer: ```suggestion
- CI tests now lint with the nf-core tools version matching the template version of the pipeline
```
